renku/ui/cli/run.py

```diff
@@ -327,43 +539,121 @@ def run(
     no_output_detection,
     success_codes,
     isolation,
+    file,
     command_line,
     verbose,
     creators,
+    dry_run,
+    no_commit,
+    provider,
 ):
     """Tracking work on a specific problem."""
-    from renku.command.run import run_command
+    from renku.command.run import run_command_line_command
     from renku.core.util.metadata import construct_creators
     from renku.ui.cli.utils.terminal import print_plan
 
     communicator = ClickCallback()
-    command = run_command()
-
-    if isolation:
-        command = command.with_git_isolation()
-
-    if creators:
-        creators, _ = construct_creators(creators)
-
-    result = (
-        command.with_communicator(communicator)
-        .build()
-        .execute(
-            name=name,
-            description=description,
-            keyword=keyword,
-            explicit_inputs=explicit_inputs,
-            explicit_outputs=explicit_outputs,
-            explicit_parameters=explicit_parameters,
-            no_output=no_output,
-            no_input_detection=no_input_detection,
-            no_output_detection=no_output_detection,
-            success_codes=success_codes,
-            command_line=command_line,
-            creators=creators,
+
+    def is_workflow_file() -> bool:
+        """Some heuristics to guess if the first argument is a workflow file or not."""
+        if file:
+            return True
+        if not command_line:
+            return False
+
+        path = Path(command_line[0])
```

Comment:
```
if len(command_line) > 1:
    return False
```
seems reasonable.

Reply:
`command_line` can have more than one element when we pass step names to only execute those steps.